### ISSUE TITLE
Cannot use own AMIs because data.user_ami returns no result

### ISSUE THREAD

**Issue by miguelaferreira** [azavea/terraform-aws-ecs-cluster#18]
Since commit 0db5e6b37ae7f54a49e3d8a48433f05f9a82d093 I haven't been able to make a plan using my own AMIs.
```
terraform plan

...

* module.ecs_cluster.module.container_service_cluster.data.aws_ami.user_ami: 1 error(s) occurred:

* module.ecs_cluster.module.container_service_cluster.data.aws_ami.user_ami: data.aws_ami.user_ami: Your query returned no results. Please change your search criteria and try again.
```
The AMI I'm passing in via `var.ami_id` is being obtained with a data query, so I'm pretty sure it exists.

I was able to fix `terraform plan` by removing the `owner-alias` filter from data.user_ami.
```
  filter {
    name   = "owner-alias"
    values = ["${var.ami_owners}"]
  }
```

Also, it seems to me that using `most_recent = true` and `image-id` filter could create problems when the passed AMI via `var.ami_id` isn't the latest.

I'll be happy to make a PR for this but would first like to hear from @hectcastro and @tnation14 if I'm missing something to make this work as is.

**Comment by tnation14** [comment]
Hi Miguel, thanks for bringing this issue to our attention. I was able to replicate the error using one of our private AMIs. It turns out that `self` (one of the values in `var.ami_owners`) isn't a valid `owner-alias`, so providing that as an argument to the AMI data source throws an error. We should be using the [`owners`](https://www.terraform.io/docs/providers/aws/d/ami.html#owners) argument. I opened #20 to fix.

>Also, it seems to me that using most_recent = true and image-id filter could create problems when the passed AMI via var.ami_id isn't the latest.

I ended up removing `most_recent` because filtering by `image-id` should only ever return a single ID. `most_recent` doesn't actually get used.
